### PR TITLE
LPD-35757 Prevent InstanceScoped JS Client Extension from being applied at any other scope

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Task: https://liferay.atlassian.net/browse/LPD-35757
Story: https://liferay.atlassian.net/browse/LPD-35637

## Motivation
There is no benefit of allowing a JS client extension, that is applied to the entire instance, to be applied to the page level. 

## Goal
Hide instance scoped cxs when a user is selecting a globalJS cx to a page.

### Steps to reproduce

1. Deploy a globalJS client extension with instance scope. You can deploy the one in the sample workspace
2. Go to `Site builder > pages > configuration > JavaScript > in page head` and assert that the client extension is not listed there.
3. Enable the feature flag LPD-30371
4. Repeat step 2

Related thread: https://liferay.slack.com/archives/C04A7NLM7L1/p1725471451574919

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

File: modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java [hunk 1]
```diff
@@ -100,26 +102,45 @@ public CET getCET(long companyId, String externalReferenceCode) {
 
 	@Override
 	public List<CET> getCETs(
-			long companyId, String keywords, String type, Pagination pagination,
-			Sort sort)
+			long companyId, boolean excludeInstanceScopedCETs, String keywords,
```
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java:105]: That way, future code don't have to call that larger method when only company and type matters.

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

File: modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethib137) [reply modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java:105]: Okay.. that sounds fine as well. Thanks.

File: modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/BaseDynamicInclude.java [hunk 2]
```diff
@@ -103,9 +101,8 @@ public void include(
 
 		try {
 			List<CET> cets = cetManager.getCETs(
-				themeDisplay.getCompanyId(), null,
-				ClientExtensionEntryConstants.TYPE_GLOBAL_JS,
-				Pagination.of(QueryUtil.ALL_POS, QueryUtil.ALL_POS), null);
+				themeDisplay.getCompanyId(),
+				ClientExtensionEntryConstants.TYPE_GLOBAL_JS);
```
Comment (ethib137) [inline modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/BaseDynamicInclude.java:105]: Hey @thiago-buarqque Isn't this the case where we want to return the instance scoped CET's too? So it should be calling the longer method?

Comment (ethib137) [reply modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/BaseDynamicInclude.java:105]: Never mind, I misunderstood what the default was.

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
@ethib137 I added an integration test for the CETManager, better than an automation one

**Comment by thiago-buarqque** [comment]
Unrelated CI errors

**Comment by thiago-buarqque** [comment]
@ethib137 @algobob is this one good to go?

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Review by ethib137 — APPROVED** [review]
This looks fine to me. It probably needs to be forwarded to frontend-infra for review though, since they own CETManager.

**Comment by thiago-buarqque** [comment]
Resent to https://github.com/liferay-frontend/liferay-portal/pull/4416

**Pull request closed without merging**